### ISSUE TITLE
'like_by_tags only' only ever likes 27 images per tag

### ISSUE THREAD

**Issue by GandalftheGUI** [InstaPy/InstaPy#1513]
## Expected Behavior
'like_by_tags' should get as many links as the user has specified

## Current Behavior
'like_by_tags only' ever likes a maximum 27 images per tag.

The 'get_links_for_tag' method repeatedly scrolls down on the image tag page
```
https://www.instagram.com/explore/tags/tourcanada/
```
to load more images for before it gathers the list of links. However it looks like instagram is removing images from the DOM that are outside the view of the user.

![image](https://user-images.githubusercontent.com/6618719/36776471-bd5298fa-1c1a-11e8-8503-b8d0de9dbf99.png)

and limiting the total number of images on the page to 27 no matter how much the user scrolls.

## Possible Solution (optional)
Modify the 'get_links_for_tag' method to gather links as it scrolls instead of at the end of the scrolling process.

## InstaPy configuration
```
session = InstaPy(username=username_input, password=passowrd_input)
session.login()
session.like_by_tags(['tourcanada'], amount=100)
```

**Comment by tomvb** [comment]
I can confirm this issue. Also present when using like_by_tags() with multiple tags.

I've applied the fix from this comment: https://github.com/timgrossmann/InstaPy/issues/1365#issuecomment-366526517 and it now works as expected again.

**Comment by Evan933** [comment]
Same here for me.

**Comment by cbyrt** [comment]
same! 
Has the master folder been updated with the new code?
This is the new code. https://gist.github.com/uluQulu/35ac02e9ef58e4a6263346e13aa0e7f2/revisions
Thank you for everyone who took time to solve this issue. Much appreciated

**Comment by GandalftheGUI** [comment]
I am using the new code, but when using more than one tag it keeps crashing. One the second tag this always happens:

```
INFO [2018-03-05 12:58:10] [iamnotian]  desired amount: 100  |  top posts [disabled]: 9  |  possible posts: 803801
INFO [2018-03-05 12:58:23] [iamnotian]  Insufficient amount of links ~ trying again: 1
INFO [2018-03-05 12:58:42] [iamnotian]  Insufficient amount of links ~ trying again: 2
INFO [2018-03-05 12:59:04] [iamnotian]  Insufficient amount of links ~ trying again: 3
INFO [2018-03-05 12:59:07] [iamnotian]  Cor! Did you send too many requests? ~ let's rest some
Traceback (most recent call last):
  File "my_example.py", line 24, in <module>
    session.like_by_tags(tags, amount=100)
  File "/Users/Doom-Mac/Documents/InstaPy/instapy/instapy.py", line 824, in like_by_tags
    skip_top_posts)
  File "/Users/Doom-Mac/Documents/InstaPy/instapy/like_util.py", line 226, in get_links_for_tag
    links.extend(get_links(browser, tag, logger, media, main_elem))
  File "/Users/Doom-Mac/Documents/InstaPy/instapy/like_util.py", line 591, in get_links
    link_elems = element.find_elements_by_tag_name('a')
  File "/usr/local/lib/python2.7/site-packages/selenium/webdriver/remote/webelement.py", line 210, in find_elements_by_tag_name
    return self.find_elements(by=By.TAG_NAME, value=name)
  File "/usr/local/lib/python2.7/site-packages/selenium/webdriver/remote/webelement.py", line 495, in find_elements
    {"using": by, "value": value})['value']
  File "/usr/local/lib/python2.7/site-packages/selenium/webdriver/remote/webelement.py", line 461, in _execute
    return self._parent.execute(command, params)
  File "/usr/local/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 236, in execute
    self.error_handler.check_response(response)
  File "/usr/local/lib/python2.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 192, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.StaleElementReferenceException: Message: stale element reference: element is not attached to the page document
  (Session info: chrome=64.0.3282.186)
  (Driver info: chromedriver=2.31.488774 (7e15618d1bf16df8bf0ecf2914ed1964a387ba0b),platform=Mac OS X 10.13.3 x86_64)
```

I am working on a fix but has anyone else run into this?

**Comment by GandalftheGUI** [comment]
I have added the new code but now I keep running into this error when it tries to load images from the second tag:
```
INFO [2018-03-05 12:58:01] [USERNAME]  Tag [2/10]
INFO [2018-03-05 12:58:01] [USERNAME]  --> mg5k
INFO [2018-03-05 12:58:10] [USERNAME]  desired amount: 100  |  top posts [disabled]: 9  |  possible posts: 803801
INFO [2018-03-05 12:58:23] [USERNAME]  Insufficient amount of links ~ trying again: 1
INFO [2018-03-05 12:58:42] [USERNAME]  Insufficient amount of links ~ trying again: 2
INFO [2018-03-05 12:59:04] [USERNAME]  Insufficient amount of links ~ trying again: 3
INFO [2018-03-05 12:59:07] [USERNAME]  Cor! Did you send too many requests? ~ let's rest some
Traceback (most recent call last):
  File "my_example.py", line 24, in <module>
    session.like_by_tags(tags, amount=100)
  File "/InstaPy/instapy/instapy.py", line 824, in like_by_tags
    skip_top_posts)
  File "/InstaPy/instapy/like_util.py", line 226, in get_links_for_tag
    links.extend(get_links(browser, tag, logger, media, main_elem))
  File "/InstaPy/instapy/like_util.py", line 591, in get_links
    link_elems = element.find_elements_by_tag_name('a')
  File "/usr/local/lib/python2.7/site-packages/selenium/webdriver/remote/webelement.py", line 210, in find_elements_by_tag_name
    return self.find_elements(by=By.TAG_NAME, value=name)
  File "/usr/local/lib/python2.7/site-packages/selenium/webdriver/remote/webelement.py", line 495, in find_elements
    {"using": by, "value": value})['value']
  File "/usr/local/lib/python2.7/site-packages/selenium/webdriver/remote/webelement.py", line 461, in _execute
    return self._parent.execute(command, params)
  File "/usr/local/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 236, in execute
    self.error_handler.check_response(response)
  File "/usr/local/lib/python2.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 192, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.StaleElementReferenceException: Message: stale element reference: element is not attached to the page document
```
I am working on a fix but was wondering if anyone else was running into a similar issue

**Comment by sivadinesh1** [comment]
Same Issue here. Desperately waiting for the solution to be commited to main branch. Please fix it.

**Comment by cbyrt** [comment]
Hi
Yes it looks there still some issues. Hopefully there will a solution soon. Thank you to everyone who is working on it.

**Comment by timgrossmann** [comment]
Should be fixed by #1844
